dashboard/app/models/follower.rb

```diff
@@ -41,4 +41,9 @@ def section_must_belong_to_teacher
   validate :cannot_follow_yourself, :teacher_must_be_teacher, :section_must_belong_to_teacher
 
   validates_presence_of :user, :student_user, :section
+
+  after_create :assign_script
+  def assign_script
+    student_user.assign_script(section.script) if section.try(:script)
```

Comment:
the try would have been to protect against the follower's section_id being nil or the section not existing -- this is checked in the validation immediately above so you're right, section can't be null. In the previous version of the code it was not as clear because the section.script check was far away from the validation...